tests/integration/optimizers/fashion/data.py

```diff
@@ -0,0 +1,60 @@
+import gzip
```

Comment:
for integration tests, isn't there a more minimal test that do not require these amount of code?

Reply:
I am not able to find anything which we can trim

Reply:
you can have a single executor with a dummy set of parameters that one can easily understand what is the best parameter set. 

You can have a DummyExecutor with 2 parameters with 3 options each for which there is only one convination that provides good results (for instance u could evaluate embedding distance) . And assert that the best set of parameters is the only possible one. That would be a good test documenting how the optimizer works and what it achieves.